tumorotek-webapp/src/main/java/fr/aphp/tumorotek/action/stats/im/model/StatResultsRow.java

```diff
@@ -126,19 +115,16 @@ public boolean equals(final Object obj){
       if(this == obj){
          return true;
       }
-      if((obj == null) || (!(obj instanceof StatResultsRow))){
+      if((!(obj instanceof StatResultsRow))){
          return false;
       }
 
       final StatResultsRow st = (StatResultsRow) obj;
 
-      return (this.subDivId == st.getSubDivId() || (subDivId != null && subDivId.equals(st.getSubDivId())))
+      return (this.subDivId.equals(st.getSubDivId()) || (subDivId != null && subDivId.equals(st.getSubDivId())))
```

Comment:
le 1er test est généralement écrit pour gérer le cas du null ou le cas où c'est la même instance d'objet. Avec la modification faite, les 2 parties du || sont exactement identiques mais la 1ere partie n'a pas le test "non null" donc elle va lancer un NullPointerException si subDivId est null. Il ne fallait donc pas toucher à cette ligne.

Reply:
je comprends mieux. Quand j'ai vu que on compare des objects avec == qui est un vrai non non en Java, je fais cette test
![image](https://github.com/TumoroteK/TumoroteK/assets/75860271/8e5f8d0e-202d-4271-b34c-a4480f661652)

Je vais pas toucher cette methode. ( Il y a peut etre une façon plus simple de le faire)

```suggestion
 return Objects.equals(this.subDivId, st.getSubDivId())
        && Objects.equals(this.banque, st.getBanque());
```